go-eicio/reader.go

```diff
@@ -157,9 +158,24 @@ func (rdr *Reader) Get() (*Event, error) {
 		err = ErrResync
 	}
 
+	rdr.Err = err
 	return event, err
 }
 
+//Events returns a channel of type Event where all of the events in the stream
+//will be pushed.  For error checking in this iteration scheme, check
+//Reader.Err.
+func (rdr *Reader) Events() <-chan *Event {
+	events := make(chan *Event)
+	go func() {
+		for event, _ := rdr.Get(); event != nil; event, _ = rdr.Get() {
+			events <- event
+		}
+		close(events)
```

Comment:
you should rather put it as a `defer(close(events))` at the top of the goroutine.

Reply:
Ah, good call!